src/libsync/propagatedownload.cpp

```diff
@@ -484,7 +486,22 @@ void PropagateDownloadFileQNAM::slotGetFinished()
         return;
     }
 
-    downloadFinished();
```

Comment:
downloadFinished() is bound to the validator-signal validated(). That is called in either case, if the checksum is not there at all, or if it is correct. Both are correct states.